docs/scripts_md/MRI.md

```diff
@@ -0,0 +1,366 @@
+string. Range string is a comma-separated list of range units. A single range
+unit follows the syntax either "X" or "X-Y".
+
+INPUT: numerical value and the range to use
+
+RETURNS: 1 if the value is in range, 0 otherwise
+
+### floats\_are\_equal($f1, $f2, $nb\_decimals)
+
+Checks whether f1 and f2 are equal (considers only the first nb\_decimals
+decimals).
+
+INPUT: float 1, float 2 and the number of first decimals
+
+RETURNS: 1 if the numbers are relatively equal, 0 otherwise
+
+### range\_to\_sql($field, $range\_string)
+
+Generates a valid SQL WHERE expression to test `$field` against
+`$range_string` using the same `$range_string` syntax as `&in_range()`.
+It returns a scalar range SQL string appropriate to use as a WHERE condition
+(`SELECT ... WHERE range_to_sql(...)`).
+
+INPUT: scalar field, scalar range string that follows the same format as in
+`&in_range()`
+
+RETURNS: scalar range SQL string
+
+### register\_db($file\_ref)
+
+Registers the NeuroDB::File object referenced by `$file_ref` into the database.
+
+INPUT: file hashref
+
+RETURNS: 0 if the file is already registered, the new FileID otherwise
+
+### mapDicomParameters($file\_ref)
+
+Maps DICOM parameters to more meaningful names in the NeuroDB::File object
+referenced by `$file_ref`.
+
+INPUT: file hashref
+
+### findScannerID($manufacturer, $model, $serialNumber, ...)
+
+Finds the scannerID for the scanner as defined by `$manufacturer`, `$model`,
+`$serialNumber`, `$softwareVersion`, using the database attached to the DBI
+database handle reference `$dbhr`. If no scannerID exists, one will be
+created.
+
+INPUT:
+  - $manufacturer   : scanner's manufacturer
+  - $model          : scanner's model
+  - $serialNumber   : scanner's serial number
+  - $softwareVersion: scanner's software version
+  - $centerID       : scanner's center ID
+  - $dbhr           : database handle reference
+  - $register\_new   : if set, will call the function `&registerScanner`
+
+RETURNS: (int) scanner ID
+
+### registerScanner($manufacturer, $model, $serialNumber, ...)
+
+Registers the scanner as defined by `$manufacturer`, `$model`,
+`$serialNumber`, `$softwareVersion`, into the database attached to the DBI
+database handle reference `$dbhr`.
+
+INPUT:
+  - $manufacturer   : scanner's manufacturer
+  - $model          : scanner's model
+  - $serialNumber   : scanner's serial number
+  - $softwareVersion: scanner's software version
+  - $centerID       : scanner's center ID
+  - $dbhr           : database handle reference
+
+RETURNS: (int) scanner ID
+
+### createNewCandID($dbhr)
+
+Creates a new CandID.
+
+INPUT: database handle reference
+
+RETURNS: (int) CandID
+
+### getPSC($patientName, $dbhr)
+
+Looks for the site alias in whatever field (usually patient\_name or
+patient\_id) is provided and return the MRI alias and CenterID.
+
+INPUT: patient name, database handle reference
+
+RETURNS: a two element array:
+  - first is the MRI alias of the PSC or "UNKN"
+  - second is the centerID or 0
```

Comment:
perhaps no need to list as bullet form? not done in this file although we have multiple outputs returned (just for consistency)